### PR TITLE
Execution unit tests

### PR DESCRIPTION
**Pull request by julianxcarter**
# Summary
Adds unit tests for the `execution` utility function. These new unit tests live in the `assurance` directory to separate them from the mCODE unit tests that are part of the actual mCODE testing harness.

This PR also establishes two new yarn commands for use in testing; `yarn translate:assurance` which converts sample CQL to ELM within the assurance directory, and `yarn test:assurance` which runs unit tests in the assurance directory. 

## New behavior 
No new behavior within the actual testing harness.

## Code changes
* The `yarn test:unit` has been changed to only run unit tests within the main `test` directory.
* The `build.js` script  now expects two optional parameters from the command line. The first argument is the path to the CQL to be translated, and the second is the path for the translated Elm to be written to. Both paths are relative to the `mcode-cql` project as a whole.
* The execution utility now includes an optional `libraryID` argument used to match on ELM objects. This argument has a default value of `mCODE`.
* `yarn test:assurance` and `yarn translate:assurance` commands have been added.

## Testing guidance
Ensure that the unit tests included for the execution utility make sense and pass. Also, if you have any opinions about the folder structure of the utility unit tests make a note of them and I can change things around!

### CODE REVIEW

**Review by mgramigna — CHANGES_REQUESTED** [review]
This looks great. I think this is set up in a very nice way for future iterations.

The only change I have is a nitpick, but this project was actually set up with yarn and not npm back in the day, so when installing we should only do `yarn install` instead of `npm install` to not generate a package-lock.json, since we already have yarn.lock. Could you just remove package-lock.json and commit the changes? Then I'm good to approve

File: test/helpers/execution.js
Comment (mgramigna) [inline test/helpers/execution.js:9]:
```suggestion
 * @param {String} libraryID the library ID of the cql library corresponding to the ELM
```

Comment (julianxcarter) [reply test/helpers/execution.js]: Commit `3eaedb6` should take care of both of your comments!

**Review by mgramigna — CHANGES_REQUESTED** [review]
Oooo crap one more thing sorry. Now that #5 is merged, if we rebase this, we can add a step in the github actions workflow to run the assurance tests as well? That way if anything breaks w/r/t the testing harness we'll know during PR.

File: README.md
Comment (Dtphelan1) [inline README.md:55]:
```suggestion
The `assurance` folder contains tests and fixtures for testing the functionality of the helper functions used in the testing harness. 
```

A few edits for clarity. Mainly, I want to avoid calling these unit tests until we properly rename the other suite of tests to be something other than `test:unit`

File: README.md
Comment (Dtphelan1) [inline README.md:61]: We might want to provide instructions for tearing down the `cql-translation-service` when people are done, since this runs the container in a detached manner. Thoughts @mgramigna

**Comment by Dtphelan1** [comment]
> Oooo crap one more thing sorry. Now that #5 is merged, if we rebase this, we can add a step in the github actions workflow to run the assurance tests as well? That way if anything breaks w/r/t the testing harness we'll know during PR.

@julianxcarter I'm sure @mgramigna and I can assist with this, if you're unfamiliar with setting up GH Actions

File: assurance/execution.test.js
Comment (Dtphelan1) [inline assurance/execution.test.js:22]: Do we need a `beforeEach` here? Do these values get changed by the blocks of testing code? If not, I believe we can use `beforeAll` to avoid loading fixtures anew before each test.

File: assurance/execution.test.js
Comment (Dtphelan1) [inline assurance/execution.test.js:49]: We should definitely feel free to add this special case to our .eslintrc. Take a look at how we defined special cases for ignoring this rule in the MEF: https://github.com/mcode/mcode-extraction-framework/blob/master/.eslintrc.json

After doing that, we should remove the few disables we have throughout this file.

File: assurance/execution.test.js
Comment (julianxcarter) [reply assurance/execution.test.js:22]: The valueSetMap is changed in only one of the unit tests, so  a `beforeAll` is probably all that's needed here. It'd be less costly to just initialize a new `valueSetMap` in that test specifically.

File: README.md
Comment (julianxcarter) [reply README.md]: I think this would be a good thing to include as well, holding off to see if Matt thinks so too though!

File: .eslintrc.js
Comment (Dtphelan1) [inline .eslintrc.js:13]: 💯

File: test/helpers/execution.js
Comment (Dtphelan1) [inline test/helpers/execution.js:9]: 💯

File: .github/workflows/ci.yml
Comment (Dtphelan1) [inline .github/workflows/ci.yml:46]: It looks like this advice has undone us, because it needs the other docker commands to be run. If you wanted, you could add the docker `start` and `stop` scripts as well. Again, thoughts @mgramigna

**Comment by julianxcarter** [comment]
Commenting to note that as of right now, example CQL within the `assurance` folder must be translated manually prior to pushing commits. A task will be made in the future to explore creating shell scripts to spin up a docker container, and translate the ELM in an automated way. Once that is done, this shell script can be called from Github Actions as well.

**Review by Dtphelan1 — APPROVED** [review]

**Comment by Dtphelan1** [comment]
Looks like this needs one more review from @mgramigna

**Review by mgramigna — APPROVED** [review]

**Comment by mgramigna** [comment]
@julianxcarter do the honors my friend

**Pull request merged**
